src/components/Form.tsx

```diff
@@ -0,0 +1,219 @@
+    const imageUrl = Boolean(data.image[0])
+      ? String(URL.createObjectURL(data.image[0] as File))
+      : '';
+
+    const photoCard: Photo = {
+      author: data.author,
+      description: data.description,
+      date: data.date,
+      category: data.category,
+      hideAuthor: data.hideAuthor,
+      human: data.human,
+      image: imageUrl,
+    };
+    setCompletedForm(true);
+    props.updateData(photoCard);
+    homeCards.push(photoCard);
+    setTimeout(() => setCompletedForm(false), 2000);
+  };
+
+  React.useEffect(() => {
+    if (isSubmitSuccessful) {
+      reset();
+    }
+  }, [reset, isSubmitSuccessful]);
+
+  return (
+    <form className="forms-container" onSubmit={handleSubmit(onSubmit)}>
+      <h2 className="forms-title-base">Upload photo</h2>
+      {сompletedForm ? (
+        <h2 className="forms-title-completed">Data saved and uploaded</h2>
+      ) : (
+        <>
+          <div className="item">
+            <label htmlFor="">Author:</label>
+            <input
+              {...register('author', {
+                minLength: {
+                  value: 2,
+                  message: 'The author must have at least 1 and no more than 20 characters.',
+                },
+                maxLength: {
+                  value: 20,
+                  message: 'The author must have at least 1 and no more than 20 characters.',
+                },
+                required: 'This field is required!',
+              })}
+              type="text"
+              name="author"
+              id="author"
+              className="author"
+              role="author-input"
+            />
+          </div>
+          {errors?.author && <p style={{ color: 'red' }}>{errors.author.message}</p>}
+          <div className="item">
+            <label htmlFor="">Description:</label>
+            <input
+              {...register('description', {
+                minLength: {
+                  value: 4,
+                  message:
+                    'The description must contain not less than 3 and not more than 40 characters.',
+                },
+                maxLength: {
+                  value: 40,
+                  message:
+                    'The description must contain not less than 3 and not more than 40 characters.',
+                },
+                required: 'This field is required!',
+              })}
+              type="text"
+              name="description"
+              id="description"
+              className="description"
+              role="description-input"
+            />
+          </div>
+          {errors?.description && <p style={{ color: 'red' }}>{errors.description.message}</p>}
+          <div className="item">
+            <label htmlFor="">Date shot:</label>
+            <input
+              {...register('date', {
+                required: 'This field is required!',
+                validate: (value) =>
+                  new Date(value) <
```

Comment:
You can write new Date(value) < new Date(). Creating Date without parameters will take the current date by default